extensions/ql-vscode/src/model-editor/generate.ts

```diff
@@ -0,0 +1,99 @@
+import { CancellationToken } from "vscode";
+import { DatabaseItem } from "../databases/local-databases";
+import { basename } from "path";
+import { QueryRunner } from "../query-server";
+import { CodeQLCliServer } from "../codeql-cli/cli";
+import { ProgressCallback } from "../common/vscode/progress";
+import { getOnDiskWorkspaceFolders } from "../common/vscode/workspace-folders";
+import { ModeledMethod } from "./modeled-method";
+import { runQuery } from "../local-queries/run-query";
+import { QueryConstraints, resolveQueries } from "../local-queries";
+import { DecodedBqrs } from "../common/bqrs-cli-types";
+type GenerateQueriesOptions = {
+  queryConstraints: QueryConstraints;
+  filterQueries?: (queryPath: string) => boolean;
+  parseResults: (
+    queryPath: string,
+    results: DecodedBqrs,
+  ) => ModeledMethod[] | Promise<ModeledMethod[]>;
+  onResults: (results: ModeledMethod[]) => void | Promise<void>;
+
+  cliServer: CodeQLCliServer;
+  queryRunner: QueryRunner;
+  queryStorageDir: string;
+  databaseItem: DatabaseItem;
+  progress: ProgressCallback;
+  token: CancellationToken;
+};
+
+export async function runGenerateQueries(options: GenerateQueriesOptions) {
+  const { queryConstraints, filterQueries, parseResults, onResults } = options;
+
+  options.progress({
+    message: "Resolving queries",
+    step: 1,
+    maxStep: 5000,
+  });
+
+  const packsToSearch = [`codeql/${options.databaseItem.language}-queries`];
+  const queryPaths = await resolveQueries(
+    options.cliServer,
+    packsToSearch,
+    "generate model",
+    queryConstraints,
+  );
+
+  const filteredQueryPaths = filterQueries
+    ? queryPaths.filter(filterQueries)
+    : queryPaths;
+
+  const maxStep = filteredQueryPaths.length * 1000;
+
+  for (let i = 0; i < filteredQueryPaths.length; i++) {
+    const queryPath = filteredQueryPaths[i];
```

Comment:
I dont think this is a problem, but would this just ignore it if one of the queries wasnt found whereas before it would error?

Reply:
Yes, that's another difference that I forgot to mention. It should still error if it doesn't find any queries, so I think this is fine.